src/UglyToad.PdfPig.Tokens/DictionaryToken.cs

```diff
@@ -114,6 +114,26 @@ public DictionaryToken With(string key, IToken value)
             return new DictionaryToken(result);
         }
 
+
+        /// <inheritdoc />
+        public bool Equals(IToken obj)
+        {
+            if (this == obj)
+                return true;
+
+            if (!(obj is DictionaryToken other))
+            {
+                return false;
+            }
+
+            if (Data.Count != other.Data.Count)
+                return false;
+
+            // TODO: Maybe consider using a sorted dictionary?
```

Comment:
Is this a good idea? 

As of right know. Every time we compare dictionary we have to sort by their key and then compare. If we use a sorted dictionary, this would only be made once per insertion. But, we would lose the order of insertion, I don't know if that is important enough?

Reply:
I would implement your alternative.